File: Shapes.VBLibrary/Circle.vb
Imports System

Namespace Shapes.VBLibrary

    Public Class Circle
        Inherits Shape

        Private _radius As Double

        Public Sub New(radius As Double)
            _radius = radius
        End Sub

        Public Property Radius As Double
            Get
                Return _radius
            End Get
            Set(value As Double)
                _radius = value
            End Set
        End Property

        Public Overrides Function Area() As Double
            Return Math.PI * Math.Pow(_radius, 2)
        End Function

        Public Function Circumference() As Double
            Return 2 * Math.PI * _radius
        End Function

    End Class

End Namespace

File: Shapes/Circle.cs
using System;

namespace Shapes
{
    public class Circle
    {
        private double radius;

        public Circle(double radius)
        {
            this.radius = radius;
        }

        public double CalculateArea()
        {
            return Math.PI * Math.Pow(this.radius, 2);
        }
    }
}
